### PR TITLE
fixed url preview for cloudflare enabled site

### LINKED ISSUE #1098: URL previews are not generated for some websites on activity page.
**Describe the bug**
A user is trying to post website URLs in the activities, some of them are porn websites. And the preview is not generated. 

I tested this on the user's site, on our demo and on my local and I could replicate the issue. I used the following two URLs shared by the user. 
- https://xhamster.com/videos/corriendo-desnudas-bajo-el-sol-5851249
- http://www.bodycontact.com/

User video - https://screenrec.com/share/CveplTSWd7

I had confirmed this with Chetan and he mentioned that there is no restriction added for porn sites. 

**To Reproduce**
1. Go to the activity page
2. Share the URLs mentioned above
3. See error

**Expected behavior**
The preview should be generated for all the URLs if there is no restriction added for porn sites. 

**Screenshots**
- https://www.loom.com/share/f913ce7ef0f94f50a2c0ab5f2f2eac38

**Support ticket links**
- https://secure.helpscout.net/conversation/1201267672/78892/
- https://secure.helpscout.net/conversation/1207629972/80336
- https://secure.helpscout.net/conversation/1273939645/97255?folderId=3955985

**Jira issue** : [PROD-940]

[PROD-940]: https://buddyboss.atlassian.net/browse/PROD-940?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by sadrul**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

user-agent set empty was creating issue, removing that line fixed the issue. I tested medium.com, cloudflare sites and various other sites.

Fixes #1098 .

### How to test the changes in this Pull Request:

1. Insert URL on activity feed form
2. Then see how it behaves

### Proof Screenshots or Video
https://www.loom.com/share/6288a35c886c43779f226c33d9587c60

### Links: 
- https://mjbizdaily.com/doj-official-reiterates-stance-against-marijuana-firms-in-280e-fight
- https://elemental.medium.com/a-supercomputer-analyzed-covid-19-and-an-interesting-new-theory-has-emerged-31cb8eba9d63

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: src/bp-core/bp-core-functions.php [hunk 1]
```diff
@@ -4936,12 +4936,7 @@ function bp_core_parse_url( $url ) {
 	} else {
 
 		// safely get URL and response body.
-		$response = wp_safe_remote_get(
-			$url,
-			array(
-				'user-agent' => '', // Default value being blocked by Cloudflare
-			)
-		);
+		$response = wp_safe_remote_get( $url );
```
Comment (KartikSuthar) [inline src/bp-core/bp-core-functions.php:4939]: @sadrul Please test after removing the code the original issue was fixed as per the PR: https://github.com/buddyboss/buddyboss-platform/pull/1520

Comment (sadrul) [reply src/bp-core/bp-core-functions.php:4939]: @KartikSuthar I tested after removing that code and found it working

**Comment by johnlorenzbb** [comment]
same issue: https://secure.helpscout.net/conversation/1572941130/153666/

**Review by chetansatasiya — APPROVED** [review]

**Comment by chetansatasiya** [comment]
@sadrul Merge in dev and add in testing.

**Comment by sadrul** [comment]
@sumitbuddyboss @mohit-buddyboss merged into dev and ready for testing

**Comment by mohit-buddyboss** [comment]
@sadrul 
Tested this on dev branch of the platform.

As discussed preview for cloudfare website is not getting generate : https://mjbizdaily.com/doj-official-reiterates-stance-against-marijuana-firms-in-280e-fight/ even after removing cache from db.

kindly look into this.

cc @sourabhmatolia @KartikSuthar @chetansatasiya

**Comment by sadrul** [comment]
This site https://mjbizdaily.com/doj-official-reiterates-stance-against-marijuana-firms-in-280e-fight/ using special protection from cloudflare. It works fine on FB, Twitter, Linkedin. Looks like they provided special provision for these big giants. But when we check Preview Generator or Online Preview then it fails. We can't fix this from our end, they need to reduce the protection from their end. So, I am closing this PR as its not a real issue.

**Comment by chetansatasiya** [comment]
@sadrul @KartikSuthar The issue is coming because the site is using the Cloudflare DDoS protection and when we are trying to scrape the URL it's first checking the Javascript & cookies to be enabled and via PHP curl it's not getting and not giving the data.

**Pull request closed without merging**